integrations/google_genai/src/haystack_integrations/components/embedders/google_genai/document_embedder.py

```diff
@@ -171,6 +171,32 @@ def _embed_batch(
 
         return all_embeddings, meta
 
+    async def _embed_batch_async(
+        self, texts_to_embed: List[str], batch_size: int
+    ) -> Tuple[List[List[float]], Dict[str, Any]]:
```

Comment:
This return type  should match the sync version, allowing for Optional[List[float]] in case of failure